lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexFuture.java

```diff
@@ -174,10 +177,10 @@ public TOutput get(final long timeout, final TimeUnit unit)
     }
   }
 
-  /**
-   * Called by VortexMaster to let the user know that the task completed.
-   */
-  public void completed(final TOutput result) {
+  @Private
+  @Override
+  public void completed(final int taskletId, final TOutput result) {
+    assert taskletId == this.futureTaskletId;
```

Comment:
Is this assertion for the current assumption that 1:1 mapping between taskletId and VortexFuture? Actually, I'm not sure how `TaskletStateDelegate` enables 1:many mapping.